RecoTracker/DeDx/plugins/DeDxEstimatorProducer.cc

```diff
@@ -110,27 +119,39 @@ void DeDxEstimatorProducer::produce(edm::Event& iEvent, const edm::EventSetup& i
 
   edm::Handle<reco::TrackCollection> trackCollectionHandle;
   iEvent.getByToken(m_tracksTag, trackCollectionHandle);
+  const auto& pixelDeDxHits = iEvent.getHandle(m_pixelDeDxHitsTag);
+  const auto& stripDeDxHits = iEvent.getHandle(m_stripDeDxHitsTag);
 
   std::vector<DeDxData> dedxEstimate(trackCollectionHandle->size());
 
   for (unsigned int j = 0; j < trackCollectionHandle->size(); j++) {
```

Comment:
minor point, but I think you could probably use an iterator here instead.
I think it would be something like:
```
     for (it = tracks.begin(); it != tracks.end(); ++it) {
       if (it->isNonnull()) {
         const reco::TrackRef track(tracks, (*it).key());
       }
     }
```

Reply:
> minor point, but I think you could probably use an iterator here instead. I think it would be something like:
> 
> ```
>      for (it = tracks.begin(); it != tracks.end(); ++it) {
>        if (it->isNonnull()) {
>          const reco::TrackRef track(tracks, (*it).key());
>        }
>      }
> ```

Just had a quick look on how implement this comment but I not sure how with current EDToken (that points to std::vector<reco::Track). Maybe changing to a edm::View<reco::Track> could allow using iterators with edm::Ref